pkg/controller/dynamic/dynamic_controller_integration_test.go

```diff
@@ -452,8 +452,9 @@ func testUpdate(ctx context.Context, t *testing.T, testContext testrunner.TestCo
 		if gcpUnstruct.Object["spec"] == nil {
 			t.Fatalf("GCP resource has a nil spec even though it was created using a resource with a non-nil spec")
 		}
-		changedFields := getChangedFields(initialUnstruct.Object, reconciledUnstruct.Object, "spec")
-		assertObjectContains(t, gcpUnstruct.Object["spec"].(map[string]interface{}), changedFields)
+		changedSpecFields := getChangedFields(initialUnstruct.Object, reconciledUnstruct.Object, "spec")
+		removeSensitiveFeilds(reconciledUnstruct, changedSpecFields) // remove sensitive fields which are reacted by the GCP API
```

Comment:
IIUC, those sensitive fields are unreadable, right? Is there any design in direct controller that captures unreadable fields and handle them differently?

Reply:
> IIUC, those sensitive fields are unreadable, right? Is there any design in direct controller that captures unreadable fields and handle them differently?

The API returns these fields with some random value. I have included an example test failure in the PR description.

Reply:
Ah, thank you for the pointer. This API design looks a bit weird. We might want to come up with a better wrap if it happens again...